### PR TITLE
Docker docs

### PR DESCRIPTION
**Pull request by phlax**
Add docs explaing dev and demo installs, and updating the README

### CODE REVIEW

File: docs/developers/docker.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
+.. _docker:
+
+Docker
+======
+
+You can install a development and test environment using docker.
+
+
+.. _docker#assumptions:
+
```
Comment (unho) [inline docs/developers/docker.rst]: Nit: Only one blank line between anchor and the related title.

File: README.rst [hunk 2]
```diff
@@ -30,16 +30,46 @@ Pootle.  Teams can manage their files, permissions, projects, and translate
...
+As the docker compose configuration is stored in the pootle directory, you will
+need ``git`` installed and ``pip`` for some minimal python dependencies to get
+set up.
+
+
+Virtualenv
+----------
+
+You can optionally use a virtualenv to install the python requirements. This
+will make it easier to install and run as a user other than ``root``
```
Comment (unho) [inline README.rst]: Nit: Missing trailing dot.

File: README.rst [hunk 3]
```diff
@@ -30,16 +30,46 @@ Pootle.  Teams can manage their files, permissions, projects, and translate
...
-Installation
+Getting started
+---------------
+
+The easiest way to get started is by using the docker demo. You will need docker
+installed on your system, and to be running as a user with permissions to manage
+docker.
+
+As the docker compose configuration is stored in the pootle directory, you will
+need ``git`` installed and ``pip`` for some minimal python dependencies to get
```
Comment (unho) [inline README.rst]: These are command so please highlight them using ``:command:`git` ``.

File: README.rst (hunk 2 above)
Comment (phlax) [reply README.rst]: do you mean a full stop ?

File: README.rst [hunk 4]
```diff
@@ -30,16 +30,46 @@ Pootle.  Teams can manage their files, permissions, projects, and translate
 on-line.  Files can be downloaded for offline translation.
 
 
-Installation
+Getting started
```
Comment (unho) [inline README.rst]: IMHO this might be confusing. Can it be renamed to something like "Trying Pootle" or something in that way?

Comment (unho) [reply README.rst]: Also, are we no longer pointing to install instructions for users that want to got directly to install instead of test run?

File: docs/developers/docker.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Assumptions
+-----------
+
+- You have docker installed and working on your system
+- The user you are using has permissions to use docker
+
+
+.. _docker#virtualenv:
+
+Create a virtualenv (optional)
```
Comment (unho) [inline docs/developers/docker.rst]: IMHO I wouldn't make it optional. When giving instructions it is better to reduce the number of optional things so when someone has to provide support it has less paths to explore.

File: docs/developers/docker.rst [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+Installing with virtualenv keeps your host environment clean, and separates
+your development environment from the host.
+
+For example:
+
+.. code-block:: console
+
+  $ mkvirtualenv pootle
+  $ cd ~/virtualenvs/pootle
```
Comment (unho) [inline docs/developers/docker.rst]: Nit: Trailing slash.

File: docs/developers/docker.rst (hunk 5 above)
Comment (phlax) [reply docs/developers/docker.rst]: i think i agree - just that many seem to ignore this.

File: docs/developers/docker.rst [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+
+.. _docker#clone:
+
+Clone the pootle repository
+---------------------------
+
+You most likely want to clone *your* fork of the pootle repository, so you
+can easily create Pull Requests for your changes.
```
Comment (unho) [inline docs/developers/docker.rst]: I would rephrase to enforce developers using their fork. But for people doing a test run having a fork in place is not mandatory.

File: docs/developers/docker.rst [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+.. code-block:: console
+
+  $ mkvirtualenv pootle
+  $ cd ~/virtualenvs/pootle
+
+
+.. _docker#clone:
+
+Clone the pootle repository
```
Comment (unho) [inline docs/developers/docker.rst]: Nit: Pootle, starting with uppercase.

File: README.rst (hunk 4 above)
Comment (phlax) [reply README.rst]: i think basically the following

- abbreviated "getting started" in README
- longer getting started here
- page/s with discussion on using dev images (running tests, building assets etc)
- page/s with discussion on running in production

its gonna take some time to get all this done tho

File: docs/developers/docker.rst [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+---------------------------
+
+You most likely want to clone *your* fork of the pootle repository, so you
+can easily create Pull Requests for your changes.
+
+.. code-block:: console
+
+  (pootle): git clone git@github.com:$USER/pootle
+  (pootle): cd pootle/
```
Comment (unho) [inline docs/developers/docker.rst]: In the console examples I think that the lines should start with `(pootle) $ `.

File: docs/developers/docker.rst [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+  (pootle): pip install -r requirements/host.txt
+
+
+.. _docker#install:
+
+Install the pootle database
+---------------------------
+
+The default installer will create a postgresql database with
```
Comment (unho) [inline docs/developers/docker.rst]: PostgreSQL

Comment (unho) [reply docs/developers/docker.rst]: Also missing trailing colon.

File: docs/developers/docker.rst [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+
+.. code-block:: console
+
+  (pootle): makey install-dev
+
+This will take some time as it loads the default projects
```
Comment (unho) [inline docs/developers/docker.rst]: Missing trailing dot.

File: docs/developers/docker.rst [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+
+.. _docker#server:
+
+Run the development server
+--------------------------
+
+Once Pootle is installed you can run the development server with
```
Comment (unho) [inline docs/developers/docker.rst]: Missing trailing colon.

File: README.rst (hunk 2 above)
Comment (unho) [reply README.rst]: Yep.

File: README.rst (hunk 4 above)
Comment (unho) [reply README.rst]: Ok, as long as the README has some links to those expanded use cases at the bottom of its "Getting started" section.

File: docs/developers/docker.rst (hunk 5 above)
Comment (unho) [reply docs/developers/docker.rst]: Then whoever is providing support can tell them "We can't help you unless you follow the instructions which are well-tested and work."

**Comment by qguv** [comment]
Hey @phlax and @unho, is there any chance of this getting merged? There are no other docs on how to set up Pootle with Docker, and this is probably the easiest way of playing with Pootle in the absence of a public test server.

**Comment by phlax** [comment]
@qguv i think probably yes.

At very least it needs to be rebased, but i think it would be good to clear at least some of the nits too.

My other concern is im not clear if these docs are (still) correct without going through and checking them.

**Comment by phlax** [comment]
ive rebased it at least - ill look at clearing some of the nits...
